controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -681,3 +938,49 @@ func diffAddresses(oldAddresses, newAddresses AddressSet) (add, remove AddressSe
 	}
 	return
 }
+
+func isTargetPortInSlice(targetPort namedPort, slicePorts []dv1beta1.EndpointPort) bool {
+	for _, p := range slicePorts {
+		switch targetPort.Type {
+		case intstr.Int:
+			if *p.Port == targetPort.IntVal {
+				return true
+			}
+		case intstr.String:
+			if *p.Name == targetPort.StrVal {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func isValidSlice(es *dv1beta1.EndpointSlice) bool {
+	if es.AddressType != dv1beta1.AddressTypeIPv4 {
+		return false
+	}
+
+	serviceName, ok := es.Labels[dv1beta1.LabelServiceName]
+	if !ok && len(es.OwnerReferences) == 0 {
+		return false
+	} else if len(es.OwnerReferences) == 0 && serviceName == "" {
+		return false
+	}
+
+	return true
+}
+
+func getEndpointSliceSVCName(es *dv1beta1.EndpointSlice) (string, error) {
```

Comment:
Could we save some work elsewhere by having this return a ServiceID instead of a string?

Reply:
Sounds like a great idea, changed!